### PR TITLE
catalog-model: implement matchEntityWithRef for client side filtering

### PR DESCRIPTION
**Pull request by freben**
Context: https://discord.com/channels/687207715902193673/687207715902193679/806441390073446410

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4375-bu4tddgbvdnlqkkezpx04v9ajhtah9zh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601a7e9314dc584fd3c18052

File: packages/catalog-model/src/entity/ref.ts [hunk 1]
```diff
@@ -196,3 +211,53 @@ export function serializeEntityRef(
...
+ * @param ref A string or compound entity ref
+ * @param context An optional context of default kind and namespace, that apply
+ *                to the ref if given
+ * @returns True if matching, false otherwise
+ */
+export function matchEntityWithRef(
+  entity: Entity,
+  ref: EntityRef | EntityName,
+  context?: EntityRefContext,
+): boolean {
```
Comment (Fox32) [inline packages/catalog-model/src/entity/ref.ts:231]: Sometimes we also have to user cases that we have to refs that we want to compare, but I don't have a good example yet (and we already have an implementation for that somewhere). But not sure if it's worth to have a mega function for all cases 😆

**Review by Rugvip — APPROVED** [review]
👍 , just some thoughts on naming

File: packages/catalog-model/src/entity/ref.ts [hunk 2]
```diff
@@ -196,3 +211,53 @@ export function serializeEntityRef(
...
+ * The matching is case insensitive, and all of kind, namespace, and name must
+ * match (after applying the optional context to the ref).
+ *
+ * @param entity The entity to match
+ * @param ref A string or compound entity ref
+ * @param context An optional context of default kind and namespace, that apply
+ *                to the ref if given
+ * @returns True if matching, false otherwise
+ */
+export function matchEntityWithRef(
```
[the remark this answers is not in the document]
Comment (freben) [reply packages/catalog-model/src/entity/ref.ts]: I'm all for it! Changed.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4375-bu4tddgbvdnlqkkezpx04v9ajhtah9zh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/601a7e9314dc584fd3c18052

**Pull request merged**